src/legacy/ui/public/filter_manager/filter_generator.js

```diff
@@ -18,15 +18,13 @@
  */
 
 import _ from 'lodash';
-import { FilterBarQueryFilterProvider } from '../filter_manager/query_filter';
 import { getPhraseScript } from '@kbn/es-query';
 
 // Adds a filter to a passed state
-export function FilterManagerProvider(Private) {
-  const queryFilter = Private(FilterBarQueryFilterProvider);
-  const filterManager = {};
+export function getFilterGenerator(queryFilter) {
```

Comment:
i think we should rather export `generateFilter` method (which doesn't need to be wrapped in a provider function).

the `.add` is confusing me, why are we not using `filterManager.addFilters` ?

Reply:
`filterGen` or as it was previously called `filterManager` has a static and old function that generates filters.
Since it contains lots of legacy logic, it deserves its own PR.